packages/ui/src/working-groups/hooks/useMyEarnings.ts

```diff
@@ -0,0 +1,40 @@
+import BN from 'bn.js'
+import { startOfMonth, subHours } from 'date-fns'
+
+import { useAccounts } from '@/accounts/hooks/useAccounts'
+import { RewardPaidEventWhereInput } from '@/common/api/queries'
+
+import { RewardPaidEventFieldsFragment, useGetRewardsQuery } from '../queries'
+
+export interface UseMyEarnings {
+  last24hours: BN | null
+  month: BN | null
+}
+
+export function useMyEarnings(): UseMyEarnings {
+  const { allAccounts } = useAccounts()
+
+  const where: RewardPaidEventWhereInput = {
+    rewardAccount_in: allAccounts.map((account) => account.address),
+    createdAt_gte: startOfMonth(Date.now()).toISOString(),
+  }
+  const { loading, data } = useGetRewardsQuery({ variables: { where } })
+
+  if (loading) {
+    return {
+      last24hours: null,
+      month: null,
+    }
+  }
```

Comment:
Other query hooks return the query's `loading` as `isLoading`; I think it would be good to return it here as well, both to stay consistent and to explicitly state what's going on in the component (`{isLoading  ? <Loading /> : <...>}`)

Reply:
For value in stats we have "-" when null or undefined passed as value, that's why I returned null.

Reply:
My problem with `null` is that it doesn't communicate much in terms of intent; it can mean that value is still loading or that it's missing altogether and one cannot really tell at a glance which one it is. I don't think it's a dealbreaker here but it does affect readability somewhat (hence the comment instead of changes request).

Reply:
I'm torn here. For other things having `isLoading` is need as we return collections in most cases. This allows checking for a loading state that will differ from an empty collection.

OTOH this returns either number or null so it is a bit consistent with `useBalance()` hooks IIRC.

I think that returning `null` in this case may be OK for now.